brut.apktool/apktool-lib/src/main/java/brut/androlib/res/decoder/ARSCDecoder.java

```diff
@@ -232,17 +232,17 @@ private ResBagValue readComplexEntry() throws IOException,
 
         ResValueFactory factory = mPkg.getValueFactory();
         Duo<Integer, ResScalarValue>[] items = new Duo[count];
-        ResValue resValue;
+        ResIntBasedValue resValue;
         int resId;
 
         for (int i = 0; i < count; i++) {
             resId = mIn.readInt();
             resValue = readValue();
 
-            try {
```

Comment:
Now unit tests pass, even though this was changed up which I thought would break the changes I patched in 8254764c6c10b42f2db7d408ae32de7d7bdf082a

This was just a pass through, I haven't pulled down the merge request to verify. I will do that today and check.